sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -52,10 +52,10 @@ object TypeCoercion {
       DecimalPrecision ::
       BooleanEquality ::
       StringToIntegralCasts ::
+      Division ::
```

Comment:
Just to be sure: are you moving the `Division` rule above the `FunctionArgumentConversion` rule in order to pass correct double arguments to functions? If so, could you add a test for this?